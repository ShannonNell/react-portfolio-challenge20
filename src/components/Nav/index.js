import React from 'react';

function Nav() {
    const tabs = ['about', 'portfolio', 'contact', 'resume'];

    function tabSelected(tab) {
        console.log(`${tab} clicked`);
    };

    return (
        <ul className="flex-row">
            {tabs.map((tab) => (
                <li
                    className="mx-1"
                    key={tab}
                >
                    <a
                        data-test-id='{tab}'
                        href={'#' + tab.toLowerCase()}
                        onClick={() => tabSelected(tab)}
                        >
                        {tab}
                    </a>
                </li>
            ))}
        </ul>
    )
};

export default Nav;