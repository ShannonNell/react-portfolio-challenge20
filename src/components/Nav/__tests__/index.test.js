// __tests__/Nav.test.js with hard coded categories
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
  it('renders', () => {
    render(<Nav></Nav>);
  });

  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Nav
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
