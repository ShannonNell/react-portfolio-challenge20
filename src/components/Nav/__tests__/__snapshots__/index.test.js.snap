// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <ul
    class="flex-row"
  >
    <li
      class="mx-1"
    >
      <a
        href="#about"
      >
        about
      </a>
    </li>
    <li
      class="mx-1"
    >
      <a
        href="#portfolio"
      >
        portfolio
      </a>
    </li>
    <li
      class="mx-1"
    >
      <a
        href="#contact"
      >
        contact
      </a>
    </li>
    <li
      class="mx-1"
    >
      <a
        href="#resume"
      >
        resume
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
